presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -165,4 +181,63 @@ public RemoteTask createRemoteTask(Session session,
                 stats,
                 isBinaryTransportEnabled);
     }
+
+    @Override
+    public void removeRemoteSource(TaskId destinationTaskId, TaskId sourceTaskId, Consumer<PrestoException> onFailure)
+    {
+        // TODO: Check whether we need the onFailure callback. Since this method is used for recoverable execution,
+        //  and if the cleanup failed, the query will fail.
```

Comment:
Is this a good way to rephrase this comment:

```
// TODO: Check whether onFailure callback is needed. Because, this method is used for recoverable execution,
// and if cleanup fails the query will fail.
```